### ISSUE TITLE
Automatic downloads

### ISSUE THREAD

**Issue by lukengda** [conrad784/docker-mediathekview-webinterface#16]
Hi, i was trying to get automatic downloads with a periodic download updates to work.

This way i could set this container image as a service that would run on my server and download all configured subscriptions whenever a new episode is airing.

Have i missed something obvious? I will describe how far i have come:

* One can configure that downloads start automatically when the filmliste is loaded and new abo downloads are found. This triggers downloads when the program is starting
* With a `docker restart <container>`, the program can be restarted and this should trigger a filmliste reload and automatic downloads. I would like to configure a cronjob that does this every night to do the periodic downloads.

Issues that i have with this approach:

* The non-removable (at least for now, see #9) update check will prevent this from working automatically as soon as an image gets outdated
* Because the latest version is not working for me (same issue as #15), this is the case at the moment already so i can't even verify my approach that i described above

Other ideas how to implement the scenario

* There is no option to trigger a filmliste refresh from command line for the running MediathekView program that i have found
* The command line flag `MediathekView -auto` is no longe available which could have also done the job maybe (https://forum.mediathekview.de/topic/2647/mv-13-3-0-auto-klappt-nicht-mehr)
* Maybe one could fork and recompile MediathekView with the Auto updater removed but this seems to be a bigger task that i can't tackle at the moment and would like to avert. 

Am i the only one with this feature request? Has anyone an idea where i could look at?

**Comment by conrad784** [comment]
Because of this I am still running 13.2.1 in a custom container with a special binary where headless and fastauto was working. I remember finding this version somewhere in the forums some time ago.
I am currently deleting the filmliste before running to download the most recent list.

**Comment by lukengda** [comment]
I am currently working on a fork that will provide a way to automate downloads with the latest version. I am willing to provide a PR back to here if this is desired.

**Comment by conrad784** [comment]
Whereas I personally like to use the version you are trying to build, I guess this is out of scope of just providing a webinterface and might not be needed by most people using this container here.
So if you are ready, please feel free to link it here, but I won't sell your work as mine. If its a 100% replacement I would just recommend people to switch and archive this project.

**Comment by lukengda** [comment]
Sorry for the reference spam... I was not aware that it would always push a new reference while i get my act together with my git setup. Take a look at the two new commits if you are interested in pulling those changes into your repository. I tested it with 13.8.0 because i ran into the same issue as #15 on my synology NAS.

If you would like me to provide an option to enable the update screen suppression in case someone uses the container manually, then i would like to propose an environment variable that needs to be set to enable the background script.

**Comment by conrad784** [comment]
I just tried it out and I think its a nice idea, but we are working against the mediathekview devs here...
E.g. now starting 13.8.0 leads to this in the running container (the initial upgrade step is skipped thanks to your extension)
![image](https://user-images.githubusercontent.com/6422533/162491228-57096cdd-b83b-4b63-852d-c2b0fc59622e.png)

**Comment by conrad784** [comment]
From what I know, the normal refresh will not really refresh the list, if its not old enough? But I would be interested in this script if this is working instead of deleting the filme.json file before running the container.

**Comment by jk-andersen** [comment]
Hej,
looking for the same solution in my setup. For now I restarted the container every night, this works as long as the version of Mediathek stays the same.

**Comment by skycaptain** [comment]
Like @lukengda, I also use MediathekView on my Synology NAS. The recent versions of Container Manager now support docker-compose, which allows for easy rebuilding and restarting through a [scheduled task](https://kb.synology.com/en-id/DSM/help/DSM/AdminCenter/system_taskscheduler?version=7) running `docker-compose restart ...` or `docker-compose up --build --pull always ...`.

Additionally, I have configured MediathekView to automatically update the Filmliste (refreshed if older than 3 hours) and initiate subscription downloads after reloading. Since I have just implemented this configuration, I don't have long-term experience with it yet. Let's see if this approach holds the test of time.
